datastore/client/datastore_client.go

```diff
@@ -30,15 +33,15 @@ func NewDatastoreClient(ipAddr string, port int) *DatastoreClient {
 	}
 	return &DatastoreClient{
 		httpClient: httpClient,
-		ipAddr: ipAddr,
-		port: port,
+		ipAddr:     ipAddr,
+		port:       port,
 	}
 }
 
 /*
 Checks if a given key Exists
 */
-func (client DatastoreClient) Exists(key string) (bool, error) {
+func (client *DatastoreClient) Exists(key string) (bool, error) {
```

Comment:
Note on this: I intentionally leave the receiver to be by-value when I don't expect a function to modify the internal state of the object. This provides extra defensiveness, so that people don't accidentally modify the internal state when they don't want to.